sonar-scanner-engine/src/main/java/org/sonar/scanner/scan/DefaultComponentTree.java

```diff
@@ -41,13 +38,8 @@ public void index(InputComponent component, InputComponent parent) {
     Preconditions.checkNotNull(component);
     Preconditions.checkNotNull(parent);
     parents.put(component, parent);
-    Set<InputComponent> list = children.get(parent);
-    if (list == null) {
-      list = new LinkedHashSet<>();
-      children.put(parent, list);
-    }
-
-    list.add(component);
+    children.putIfAbsent(parent, new LinkedHashSet<>());
+    children.get(parent).add(component);
```

Comment:
If the entry is missing it will put a new `LinkedHashSet`. It will have the same outcome as what you wrote, but in case the entry is present, it won't create a new `LinkedHashSet` unnecessarily.